milli/src/vector/mod.rs

```diff
@@ -26,6 +30,171 @@ pub type Embedding = Vec<f32>;
 
 pub const REQUEST_PARALLELISM: usize = 40;
 
+pub struct ArroyWrapper {
```

Comment:
If we're going to have a struct that abstracts over the quantization, should we also abstract about multiple vectors?

i.e. this could take an embedding_id instead of an index and expose an interface to get/put multiple vectors, encapsulating the arroy_dbs iterators from the Index.